crates/control/src/role_assignment.rs

```diff
@@ -320,13 +351,20 @@ impl RoleAssignment {
         if let Some(forced_role) = context.forced_role {
             self.role = *forced_role;
         } else {
-            self.role = role;
+            self.role = new_role;
         }
         self.team_ball = team_ball;
 
         if let Some(game_controller_state) = context.filtered_game_controller_state {
-            if game_controller_state.penalties.one.is_some() {
-                self.last_time_keeper_penalized = Some(cycle_start_time);
+            for player in self
+                .last_time_player_was_penalized
+                .clone()
+                .iter()
+                .map(|(playernumber, _systemtime)| playernumber)
```

Comment:
_systemtime is not very descriptive. What do you think of _last_time_penalized?

Reply:
or even just `, _)` or `, ..)`? It is not used anyway?!

Reply:
Done